src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/BindingSourceDesigner.cs

```diff
@@ -0,0 +1,119 @@
+    ///
+    ///  In response to this action, we will send a notification to the DataSourceProviderService
+    ///  once the corresponding ComponentChanged event comes through (indicating that the new
+    ///  binding has actually taken effect).
+    ///
+    ///  This notification will cause the DataSourceProviderService to re-examine the data source
+    ///  that the BindingSource is now bound to, so it can generate any table adapters or fill
+    ///  statements necessary to properly set up that data source.
+    ///
+    ///  This serves the scenario of an advanced user configuring a BindingSource manually. In
+    ///  normal user scenarios, BindingSources get auto-generated as a result of various drag/drop
+    ///  operations, and in those scenarios the above notification is sent automatically.
+    ///
+    ///  VSWhidbey#256272
+    /// </summary>
+    public bool BindingUpdatedByUser
+    {
+        set
+        {
+            _bindingUpdatedByUser = value;
+        }
+    }
+
+    /// <summary>
+    ///  Initialize this designer.
+    /// </summary>
+    public override void Initialize(IComponent component)
+    {
+        base.Initialize(component);
+
+        IComponentChangeService componentChangeSvc = (IComponentChangeService)GetService(typeof(IComponentChangeService));
+        if (componentChangeSvc is not null)
+        {
+            componentChangeSvc.ComponentChanged += new ComponentChangedEventHandler(OnComponentChanged);
+            componentChangeSvc.ComponentRemoving += new ComponentEventHandler(OnComponentRemoving);
+        }
+    }
+
+    /// <summary>
+    ///  Dispose this designer.
+    /// </summary>
+    protected override void Dispose(bool disposing)
+    {
+        if (disposing)
+        {
+            IComponentChangeService componentChangeSvc = (IComponentChangeService)GetService(typeof(IComponentChangeService));
+            if (componentChangeSvc is not null)
+            {
+                componentChangeSvc.ComponentChanged -= new ComponentChangedEventHandler(OnComponentChanged);
+                componentChangeSvc.ComponentRemoving -= new ComponentEventHandler(OnComponentRemoving);
+            }
+        }
+
+        base.Dispose(disposing);
+    }
+
+    /// <summary>
+    ///  Spot when a change to the BindingSource's DataSource or DataMember is made by the user, and notify the
+    ///  DataSourceProviderService accordingly. See description of BindingUpdatedByUser property for full details.
+    /// </summary>
+    private void OnComponentChanged(object? sender, ComponentChangedEventArgs e)
+    {
+        if (_bindingUpdatedByUser &&
+            e.Component == this.Component &&
+            e.Member is not null && (e.Member.Name == "DataSource" || e.Member.Name == "DataMember"))
+        {
+            _bindingUpdatedByUser = false;
+
+            DataSourceProviderService dspSvc = (DataSourceProviderService)GetService(typeof(DataSourceProviderService));
```

Comment:
Please don't use abbreviations in variable names.
```suggestion
            var dataSourceProviderService = (DataSourceProviderService)GetService(typeof(DataSourceProviderService));
```